afew/MailMover.py

```diff
@@ -58,44 +61,78 @@ def move(self, maildir, rules):
             messages = notmuch.Query(self.db, main_query).search_messages()
             for message in messages:
                 # a single message (identified by Message-ID) can be in several
-                # places; only touch the one(s) that exists in this maildir 
+                # places; only touch the one(s) that exists in this maildir
                 all_message_fnames = message.get_filenames()
-                to_move_fnames = [name for name in all_message_fnames
-                                  if maildir in name]
+                to_move_fnames = [
+                    name for name in all_message_fnames
+                    if maildir in name]
                 if not to_move_fnames:
                     continue
-                moved = True
-                self.__log_move_action(message, maildir, rules[query],
-                                       self.dry_run)
                 for fname in to_move_fnames:
                     if self.dry_run:
                         continue
-                    try:
-                        shutil.copy2(fname, self.get_new_name(fname, destination))
-                        to_delete_fnames.append(fname)
-                    except shutil.SameFileError:
-                        logging.warn("trying to move '{}' onto itself".format(fname))
-                        continue
-                    except shutil.Error as e:
-                        # this is ugly, but shutil does not provide more
-                        # finely individuated errors
-                        if str(e).endswith("already exists"):
-                            continue
-                        else:
-                            raise
-
-        # remove mail from source locations only after all copies are finished
-        for fname in set(to_delete_fnames):
-            os.remove(fname)
-
-        # update notmuch database
-        if not self.dry_run:
-            if moved:
-                logging.info("updating database")
-                self.__update_db(maildir)
+                    else:
+                        moved = self.moveMail(fname, destination, True)
+        # Return the moved flag
+        return moved
+
+    """ Action of moving a mail and upgrade the DB """
+    def moveMail(self, mailFrom, mailTo, upgradeDatabase):
+        output = False
+        """ We need to check that we are not moving a file to itself """
+        if mailFrom == mailTo:
```

Comment:
Yes sure I will rebase the branch and change that.

Reply:
Hey,
after rebasing the branch, and re-reading I aggree we need to be sure that:
* mailFrom is a file AND mailTo is a directory

If this is not I will raise an error.

Reply:
Did you see those:
```
            try:
                " mailFrom must be a file "
                if not os.path.isfile(mailFrom):
                    logging.error("mailFrom not a file: {}".format(mailFrom))
                    raise SystemExit
                " mailTo must be a directory "
                if not os.path.isdir(mailTo):
                    logging.error("mailTo not a dir: {}".format(mailTo))
                    raise SystemExit
```
So those tests already exist (sorry took me a second reading to be sure),
so duplicating those is not a good idea, and moving them end in the same state.

So tell me @rpuntaie

Reply:
It seems like the commits I referred to have gone. There is not `mailFrom` anywhere anymore. So this seems to be obsolete.

As commented further below: tests are absolutely needed, to avoid break established behavior. E.g. I've seen `{}/{}/cur/` is now `{}/{}`, which is an interface change: The user would have to change its config.